### PR TITLE
update azure throttling README

### PR DESCRIPTION
**Pull request by marwanad**
Update document to add a section on throttling and document `AZURE_VMSS_CACHE_TTL` variable.

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/azure/README.md [hunk 1]
```diff
@@ -128,6 +128,24 @@ To run a cluster autoscaler pod on a master node, the deployment should tolerate
 
 To run a cluster autoscaler pod with Azure managed service identity (MSI), use [cluster-autoscaler-vmss-msi.yaml](examples/cluster-autoscaler-vmss-msi.yaml) instead.
 
+#### Azure API Throttling
+Azure has hard limits on the number of read and write requests against Azure APIs *per subscription, per region*. Running lots of clusters in a single subscription, or running a single large, dynamic cluster in a subscription can produce side effects that exceed the number of calls permitted within a given time window for a particular category of requests. See the following documents for more detail on Azure API throttling in general:
```
Comment (marwanad) [inline cluster-autoscaler/cloudprovider/azure/README.md:132]: Intro borrowed from Jack's: https://github.com/Azure/aks-engine/blob/master/docs/topics/azure-api-throttling.md#overview

File: cluster-autoscaler/cloudprovider/azure/README.md [hunk 2]
```diff
@@ -128,6 +128,26 @@ To run a cluster autoscaler pod on a master node, the deployment should tolerate
...
+- https://docs.microsoft.com/en-us/azure/virtual-machines/troubleshooting/troubleshooting-throttling-errors
+
+Given the dynamic nature of cluster autoscaler, it can be a trigger for hitting those rate limits on the subscriptions. This in turn can affect other components running in the cluster that depend on Azure APIs such as kube-controller-manager.
+
+When using K8s versions older than v1.18, we recommend using at least **v.1.17.5, v1.16.9, v1.15.12** which include various improvements on the cloud-provider side that have an impact on the number of API calls during scale down operations.
+
+As for CA versions older than 1.18, we recommend using at least **v.1.17.2, v1.16.5, v1.15.6**.
+
+In addition, cluster-autoscaler exposes a `AZURE_VMSS_CACHE_TTL` environment variable which controls the rate of `GetVMScaleSet` being made. By default, this is 15 seconds but setting this to a higher value such as 60 seconds can protected against API throttling. The caches used are proactively incremented and decremented with the scale up and down operations and this higher value doesn't have any noticeable impact on performance. **Note that the value is in seconds**
```
Comment (aramase) [inline cluster-autoscaler/cloudprovider/azure/README.md:143]: nit: s/protected/protect

Comment (marwanad) [reply cluster-autoscaler/cloudprovider/azure/README.md]: fixed, ty

**Comment by nilo19** [comment]
thanks for caring about this.
/lgtm
/approve

**Pull request merged**